xbmc/pvr/epg/EpgInfoTag.cpp

```diff
@@ -23,6 +23,11 @@
 #include "utils/Variant.h"
 #include "utils/log.h"
 
+#include "../channels/PVRChannelGroupsContainer.h"
+#include "../channels/PVRChannelGroups.h"
+#include "../channels/PVRChannelGroup.h"
+#include "../channels/PVRChannel.h"
```

Comment:
EPG must not depend on channels. This can lead to deadlocks.

Recordings -> Timers -> Channels -> EPG is the general call order.